libd-llvm/src/d/llvm/type.d

```diff
@@ -302,7 +302,19 @@ final class TypeGen {
 	}
 	
 	LLVMTypeRef visit(Interface i) {
-		assert(0, "codegen for interface is not implemented.");
+		if(auto it = i in typeSymbols) {
+			return *it;
+		}
+		auto llvmStruct = LLVMStructCreateNamed(llvmCtx, i.mangle.toStringz());
+		typeSymbols[i] = llvmStruct;
+		auto vtblStruct = LLVMStructCreateNamed(llvmCtx, (i.mangle ~ "__vtbl").toStringz());
+		auto vtblPtr = LLVMPointerType(vtblStruct, 0);
+		auto objectPtr = visit(pass.object.getObject());
+		LLVMTypeRef[2] elements;
+		elements[0] = objectPtr;
+		elements[1] = vtblPtr;
```

Comment:
This is overly verbose. You create a ton of variable to reassign them somewhere else. Just assign directly to where they are supposed to go.